src/utils/word-api.ts

```diff
@@ -101,35 +93,51 @@ class WordApi {
     return `JABREF-CITATION-${JSON.stringify(citation)}`;
   };
 
-  static async getTotalNumberOfCitations(): Promise<number | null> {
-    const jabRefCitations =
-      (await WordApi.getJabRefCitations()) as Array<Word.ContentControl>;
-    return jabRefCitations.length;
+  static async getTotalNumberOfCitations(): Promise<number | void> {
+    return Word.run(async (context) => {
+      const jabRefCitations = await WordApi.getJabRefCitations(context);
+      return jabRefCitations.length;
+    }).catch((error) => {
+      console.log(`Error: ${JSON.stringify(error)}`);
+      if (error instanceof OfficeExtension.Error) {
+        console.log(`Debug info: ${JSON.stringify(error.debugInfo)}`);
+      }
+    });
   }
 
   static async getCitationIdToPos(): Promise<Record<string, number> | void> {
-    const jabRefCitations =
-      (await WordApi.getJabRefCitations()) as Array<Word.ContentControl>;
-    const citationIdToPos: Record<string, number> = {};
-    let pos = 0;
-    jabRefCitations.forEach((citation) => {
-      const { tag } = citation;
-      citationIdToPos[tag.substring(16)] = pos;
-      pos += 1;
+    return Word.run(async (context) => {
+      const jabRefCitations = await WordApi.getJabRefCitations(context);
+      const citationIdToPos: Record<string, number> = {};
+      jabRefCitations.forEach((citation, index) => {
+        const { tag } = citation;
+        citationIdToPos[tag.substring(16)] = index;
```

Comment:
Can you extract the magic number 16 to a constant and give it a name so that it's clear what or why it's used